src/groupByKey.ts

```diff
@@ -2,6 +2,17 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T) : GroupsMap<T> {
+  return items.reduce((map:GroupsMap<T>, item: T) => {
+    const groupField = item[key] as string;
+    const groupObjectCopy: GroupsMap<T> = { ...map };
+
+    if (Object.prototype.hasOwnProperty.call(groupObjectCopy, groupField)) {
+      groupObjectCopy[groupField].push(item);
+    } else {
+      groupObjectCopy[groupField] = [item];
+    }
```

Comment:
You can remove else statement like this:

```suggestion
    if (!Object.prototype.hasOwnProperty.call(groupObjectCopy, groupField)) {
      groupObjectCopy[groupField] = [];
    }
    
    groupObjectCopy[groupField].push(item);
```